src/components/PostDetails.tsx

```diff
@@ -1,106 +1,96 @@
-import React from 'react';
+import { FC, useEffect, useState } from 'react';
+
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { CommentsList } from './CommentsList';
+
+import { deleteComment, getComments } from '../api/comments';
+
+interface Props {
+  selectedPost: Post;
+}
+
+export const PostDetails: FC<Props> = ({ selectedPost }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoadingComments, setIsLoadingComments] = useState(false);
+  const [error, setError] = useState('');
+  const [isShownForm, setIsShownForm] = useState(false);
+
+  const handleDeleteComment = (id: Comment['id']) => {
+    deleteComment(id)
+      .then(() =>
+        setComments(currentComments =>
+          currentComments.filter(comment => comment.id !== id),
+        ),
+      )
+      .catch(() => 'Unable to delete post');
```

Comment:
The catch block for `deleteComment` should handle errors more explicitly. Consider displaying an error message to the user or logging the error for debugging purposes.